pkg/networkservice/metrics/server.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metrics - implement vpp based metrics collector service, it update connection on passing Request() with set of new metrics received during interval
+package metrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.ligato.io/vpp-agent/v3/proto/ligato/configurator"
+	vpp_interfaces "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/interfaces"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/vppagent"
+)
+
+// Server - helper interface to get some various information required for testing.
+type Server interface {
+	// GetMetricsReceived - show how many events we receive from internal vpp agent inerface
+	GetMetricsReceived() int
+	networkservice.NetworkServiceServer
+}
+
+type metricsServer struct {
+	interval        time.Duration
+	connections     map[string]*connectionInfo
+	vppConnections  map[string]string
+	executor        serialize.Executor
+	vppClient       configurator.StatsPollerServiceClient
+	metricsRecieved int
+	collectorCancel context.CancelFunc
+	Server
+}
+
+type connectionInfo struct {
+	connectionID string
+	index        uint32
+	metrics      map[string]string
+	ifName       string
+}
+
+func (s *metricsServer) GetMetricsReceived() (result int) {
```

Comment:
Why not just set the interval timer in NewServer short and send a new Request through to see if you have stats rather than exposing the internals of the chain element for testing?

Reply:
Ok, will do.

Reply:
Since no timer required, resolving comment.